addons/website_sale_coupon/models/sale_coupon_program.py

```diff
@@ -34,3 +34,6 @@ def _check_promo_code(self, order, coupon_code):
         if self.website_id and self.website_id != order.website_id:
             return {'error': 'This promo code is not valid on this website.'}
         return super()._check_promo_code(order, coupon_code)
+
+    def _get_program(self, order, coupon_code):
+        return super()._get_program(order, coupon_code).filtered(lambda p: p.website_id.id in [False, order.website_id.id])
```

Comment:
Filtering on a stored field right after a search is weird, you should construct the right domain in/for `_get_program` instead.